test/convert_to_iso_string.test.ts

```diff
@@ -29,14 +30,14 @@ describe("convertToIsoString", () => {
   it("should convert moment objects to ISO strings in UTC", () => {
     const tzDate = "2023-01-01T01:00:00+01:00";
     const isoDate = "2023-01-01T00:00:00.000Z";
-    const converted = convertToIsoString(moment(tzDate));
+    const converted = convertToIsoString(moment(tzDate).toDate());
     expect(converted).toEqual(isoDate);
   });
 
   it("should convert dayjs objects to ISO strings", () => {
     const tzDate = "2023-01-01T01:00:00+01:00";
     const isoDate = "2023-01-01T00:00:00.000Z";
-    const converted = convertToIsoString(dayjs(tzDate));
+    const converted = convertToIsoString(dayjs(tzDate).toDate());
```

Comment:
The test is if it correctly converts dayjs objects, so should not be made into a date. The issue is with the typing in `convertToIsoString`